frontend/src/metabase/core/components/Slider/Slider.tsx

```diff
@@ -18,76 +22,121 @@ export type NumericInputAttributes = Omit<
 >;
 
 export interface SliderProps extends NumericInputAttributes {
-  value: number[];
-  onChange: (value: number[]) => void;
+  value: (number | undefined)[];
+  onChange: (value: (number | undefined)[]) => void;
   min?: number;
   max?: number;
   step?: number;
 }
 
 const Slider = ({
-  value,
+  value: parentValue,
   onChange,
-  min = 0,
-  max = 100,
+  min: parentMin = 0,
+  max: parentMax = 100,
   step = 1,
 }: SliderProps) => {
-  const [rangeMin, rangeMax] = useMemo(
-    () => [Math.min(...value, min, max), Math.max(...value, min, max)],
-    [value, min, max],
-  );
+  const [isHovering, setIsHovering] = useState(false);
+  const [value, setValue] = useState([
+    parentValue[0] ?? parentMin,
+    parentValue[1] ?? parentMax,
+  ]);
 
-  const [beforeRange, rangeWidth] = useMemo(() => {
-    const totalRange = rangeMax - rangeMin;
+  // potentially expand the input range to include out-of-range values
+  const [min, max] = useMemo(() => {
+    return [_.min([...value, parentMin]), _.max([...value, parentMax])];
+  }, [value, parentMin, parentMax]);
+
+  useEffect(() => {
+    setValue([parentValue[0] ?? min, parentValue[1] ?? max]);
+  }, [parentValue, min, max]);
 
+  const [beforeRange, rangeWidth] = useMemo(() => {
+    const totalRange = max - min;
     return [
-      ((Math.min(...value) - rangeMin) / totalRange) * 100,
+      ((Math.min(...value) - min) / totalRange) * 100,
       (Math.abs(value[1] - value[0]) / totalRange) * 100,
     ];
-  }, [value, rangeMin, rangeMax]);
+  }, [value, min, max]);
 
-  const handleChange = useCallback(
+  const handleInput = useCallback(
     (event: ChangeEvent<HTMLInputElement>, valueIndex: number) => {
       const changedValue = [...value];
       changedValue[valueIndex] = Number(event.target.value);
-      onChange(changedValue);
+      setValue(changedValue);
     },
-    [value, onChange],
+    [value, setValue],
   );
 
-  const sortValues = useCallback(() => {
-    if (value[0] > value[1]) {
-      const sortedValues = [...value].sort();
-      onChange(sortedValues);
-    }
+  const handleChange = useCallback(() => {
+    const sortedValues = value[1] < value[0] ? [...value].sort() : value;
+    onChange(sortedValues);
   }, [value, onChange]);
```

Comment:
we only pass this component's values up to the parent component on `mouseUp` or `keyUp` to keep the number of updates reasonable.